src/kakarot/execution_context.cairo

```diff
@@ -710,33 +711,29 @@ namespace ExecutionContext {
         );
     }
 
-    // @notice Dump the current execution context.
-    // @dev The execution context is dumped to the debug server if `DEBUG` environment variable is set to `True`.
-    // @param self The pointer to the execution context.
-    func dump{range_check_ptr}(self: model.ExecutionContext*) {
-        let pc = self.program_counter;
-        let stopped = is_stopped(self);
-
-        return ();
-    }
-
     // @notice Update the program counter.
     // @dev The program counter is updated to a given value. This is only ever called by JUMP or JUMPI
     // @param self The pointer to the execution context.
     // @param new_pc_offset The value to update the program counter by.
     // @return ExecutionContext The pointer to the updated execution context.
-    func update_program_counter{range_check_ptr}(
+    func jump{range_check_ptr}(
         self: model.ExecutionContext*, new_pc_offset: felt
     ) -> model.ExecutionContext* {
         alloc_locals;
-        // Revert if new_value points outside of the code range
-        with_attr error_message("Kakarot: new pc target out of range") {
-            assert_nn(new_pc_offset);
-            assert_le(new_pc_offset, self.call_context.bytecode_len - 1);
+
+        let is_nn_pc = is_nn(new_pc_offset);
+        let is_le_bytecode_len = is_le(new_pc_offset, self.call_context.bytecode_len - 1);
+        if (is_nn_pc + is_le_bytecode_len != 2) {
+            let (revert_reason_len, revert_reason) = Errors.programCounterOutOfRange();
+            let ctx = ExecutionContext.revert(self, revert_reason, revert_reason_len);
+            return ctx;
         }
 
-        // Revert if new pc_offset points to something other then JUMPDEST
-        check_jumpdest(self=self, pc_location=new_pc_offset);
+        if ([self.call_context.bytecode + new_pc_offset] != 0x5b) {
```

Comment:
we might want to add a TODO here, to remind us to check that the 0x5b condition checking must also include checking 0x5b does not come from a PUSH instruction

Reply:
There is an issue for this already existing in the repo